libgalois/include/katana/PropertyGraph.h

```diff
@@ -29,12 +29,81 @@ struct KATANA_EXPORT GraphTopology {
   using edges_range = StandardRange<edge_iterator>;
   using iterator = node_iterator;
 
+private:
+  LargeArray<Edge> adj_indices_;
+  LargeArray<Node> dests_;
+
+  // TODO(amber): make these private in the near future. No other class should
+  // access these directly. Instead provide access methods
+public:
   std::shared_ptr<arrow::UInt64Array> out_indices;
   std::shared_ptr<arrow::UInt32Array> out_dests;
 
-  uint64_t num_nodes() const { return out_indices ? out_indices->length() : 0; }
+private:
+  // TODO: generalize to typename T
+  static std::shared_ptr<arrow::UInt64Array> largeToArrowArray(
+      LargeArray<uint64_t>& lgArr) noexcept {
```

Comment:
`const LargeArray`

Reply:
Can't do const due to compatibility with existing usage (e.g. in Python).